apps/issuer/api.py

```diff
@@ -143,6 +196,41 @@ def post(self, request, **kwargs):
         return super(AllBadgeClassesList, self).post(request, **kwargs)
 
 
+class AllCollectionBadgeClassesList(UncachedPaginatedViewMixin, BaseEntityListView):
+    """
+    GET a list of collectionbadgeclasses within one issuer context or
+    POST to create a new collectionbadgeclass within the issuer context
+    """
+    model = CollectionBadgeContainer
+    permission_classes = [
+        IsServerAdmin
+        | (AuthenticatedWithVerifiedIdentifier & IsEditor & BadgrOAuthTokenHasScope)
+        | BadgrOAuthTokenHasEntityScope
+    ]
+    v1_serializer_class = CollectionBadgeClassSerializerV1
+    v2_serializer_class = CollectionBadgeClassSerializerV2
+    valid_scopes = ["rw:issuer"]
+
+    def get_queryset(self, request, **kwargs):
+        return CollectionBadgeContainer.objects.all().order_by('created_at')
+
+    @apispec_list_operation('CollectionBadgeClass',
+        summary="Get a list of CollectionBadgeClasses for authenticated user",
+        tags=["CollectionBadgeClasses"],
+    )
+    def get(self, request, **kwargs):
+        return super(AllCollectionBadgeClassesList, self).get(request, **kwargs)
+
+    @apispec_post_operation('CollectionBadgeClass',
+        summary="Create a new CollectionBadgeClass",
+        tags=["CollectionBadgeClasses"],
+        # TODO: Add parameters
```

Comment:
Is this TODO still up to date?

Reply:
yes

Reply:
And should it be done before being merged?

Reply:
Actually, it's not really needed for now i guess. I removed it for now